### PR TITLE
OTA-1007: Enable pre-merge test for cincinnati

### PR DESCRIPTION
**Pull request by shellyyang1989**
Per https://redhat-internal.slack.com/archives/CH76YSYSC/p1691119535303039, adding `operator` stanza to enable PR pre-merge test for cincinnati.

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
Please run `make update` to update all related jobs.

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (jianzhangbjz) [inline ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:35]: I didn't see this file under https://github.com/openshift/cincinnati-operator/tree/master/config

**Review by jianzhangbjz — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:17]: Why 4.10?

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:44]: I don't think we will need `promotion`

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (jianzhangbjz) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:17]: The index image doesn't support `SQLite` since OCP 4.11, so we need to use 4.10 here.

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (jianzhangbjz) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:44]: Yes, but if we don't promote it, it will be removed once the cluster is removed(around 2 hours). Sometimes, we need to create the operator image by using the cluster-bot, but test it in our local env. So, in case of this image is removed after 2 hours, it's better to promote it.

Comment (petr-muller) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:44]: Promotion does not play any role in cluster-bot runs, which is what this PR aims to enable. Promotion setup simply performs post-merge image builds that are promoted to a persistent location.

Bundles (and indices) are not promoteable anyway, so we would only persist actual operator image, which *could* be useful, but we do not definitely want to promote it into `ocp` namespace. Let's omit this from this PR and only enable once we have an actual use to have that image persisted.

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: I don't think we need this. Cincinnati (not the operator) is already built after each commit and pushed to `quay.io/app-sre/cincinnati`

**Comment by petr-muller** [comment]
Doing some experiments in https://github.com/openshift/release/pull/42113 and https://github.com/openshift/cincinnati-operator/tree/bundle-dockerfile

Maybe we'll need `promotion` after all but not to `ocp`

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (jianzhangbjz) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:17]: Yeah, thanks! @shellyyang1989 let's have a try!

**Comment by jianzhangbjz** [comment]
> Maybe we'll need promotion after all but not to ocp

Thanks! I guess we can promote it to the `ci` namespace.

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (jianzhangbjz) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: Actually, she should promote the `cincinnati` operand image here, such as
```console
- dockerfile_path: dist/Dockerfile.deploy/Dockerfile
  inputs:
    ubi:
      as:
      - registry.access.redhat.com/ubi8/ubi:latest
  to: cincinnati
```
And then, refer to this operand image in the operator, like
```console
  driver-operand:
    name: "4.14"
    namespace: ocp
    tag: cincinnati
```

Comment (jianzhangbjz) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: Of course, we'd better to use the `ci` namespace instead.

Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: > Cincinnati (not the operator) is already built after each commit and pushed to quay.io/app-sre/cincinnati

Is the image built before each o./cincinnati PR merge or post merge?

**Comment by petr-muller** [comment]
> Thanks! I guess we can promote it to the ci namespace.

No, rather something ours, like `cincinnati`

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (petr-muller) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: `quay.io/app-sre/cincinnati` is built after merge - by a different pipelline (app-interface), not ci-operator/openshift ci. Annoying thing I have discovered yesterday is that it gets built and pushed but there's no floating `latest` tag which we need. I'll see if we can configure app-interface to create that tag as well.

Of course all images are also built before merge by ci-operator/openshift ci. My point is that to enable pre-merge testing we don't need `promotion`.

Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: Sometimes we might enable some new features in cincinnati but delay cincinnati-operator changes as maybe we don't have bandwidth. In this case, to pre-merge test cincinnati changes, we need to use the upstream way https://github.com/openshift/cincinnati/blob/master/docs/user/running-cincinnati.md which requires cincinnati image. So imo cincinnati promotion is good to help us pre-merge test cincinnati. Please correct me if I'm wrong:)

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Thank you Petr and Jian. Removed the deprecated items.

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Thank you Petr and Jian. Dropped.

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (petr-muller) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: Promotion means "build images **after** merge and persist them": same as app-interface (except that one little detail of missing `latest` tag). 

To pre-merge test cincinnati changes, you have a pending cincinnati PR and need to build cincinnati image from the PR code. That already happens and is not related to promotion.

Comment (petr-muller) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml:54]: Also, it seems that we have some cincinnati image persisted in here (no idea how it gets there without promotion)

```
  cincinnati_operand_latest:
    name: cincinnati-build-root
    namespace: cincinnati-ci
    tag: deploy
```

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (petr-muller) [inline ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml:29]: No need to have this line, `.` is the default

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml]: Thanks for the explanation!

I just promoted the `cincinnati-ci/cincinnati-build-root:deploy` as it's referenced by the bundle image https://github.com/openshift/release/pull/42053/files#diff-3e4c9ff59bb36d000ea248cbef83680065c76329d216d38ce61c00fb798d3e94R33. Not sure if it makes sense:)

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Updated. Thanks!

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Dropped. Thanks!

File: ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml]: `cincinnati-ci/cincinnati-build-root:deploy` is promoted here: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cincinnati/openshift-cincinnati-master__customrust.yaml#L48-L50

Comment (petr-muller) [reply ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml]: Right, there's a promoting variant...

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Thank you Petr, updated.

File: ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml
Comment (shellyyang1989) [reply ci-operator/config/openshift/cincinnati-operator/openshift-cincinnati-operator-master.yaml]: Thank you, updated.

**Comment by shellyyang1989** [comment]
/pj-rehearse pull-ci-openshift-cincinnati-operator-master-ci-bundle-cincinnati-bundle pull-ci-openshift-cincinnati-operator-master-install-bundle

**Comment by shellyyang1989** [comment]
bundle.Dockerfile is being enabled here https://github.com/openshift/cincinnati-operator/pull/173

**Comment by petr-muller** [comment]
/pj-rehearse

https://github.com/openshift/cincinnati-operator/pull/173 merged :shipit:

**Comment by petr-muller** [comment]
[install-bundle](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/42053/rehearse-42053-pull-ci-openshift-cincinnati-operator-master-install-bundle/1689705622007713792/artifacts/install-bundle/install/build-log.txt) passed 🎉 

```
namespace/install-osus-here created
time="2023-08-10T18:36:20Z" level=info msg="Creating a File-Based Catalog of the bundle \"registry.build05.ci.openshift.org/ci-op-scth0tj4/pipeline@sha256:659464ca4855a1809652ac34678424b0f94357e301ee28528c44d6d36bc53c84\""
time="2023-08-10T18:36:20Z" level=info msg="Generated a valid File-Based Catalog"
time="2023-08-10T18:36:25Z" level=info msg="Created registry pod: 59464ca4855a1809652ac34678424b0f94357e301ee28528c44d6d36bc53c84"
time="2023-08-10T18:36:25Z" level=info msg="Created CatalogSource: update-service-operator-catalog"
time="2023-08-10T18:36:25Z" level=info msg="OperatorGroup \"operator-sdk-og\" created"
time="2023-08-10T18:36:25Z" level=info msg="Created Subscription: update-service-operator-v5-0-2-dev-sub"
time="2023-08-10T18:36:30Z" level=info msg="Approved InstallPlan install-nt5bn for the Subscription: update-service-operator-v5-0-2-dev-sub"
time="2023-08-10T18:36:30Z" level=info msg="Waiting for ClusterServiceVersion \"install-osus-here/update-service-operator.v5.0.2-dev\" to reach 'Succeeded' phase"
time="2023-08-10T18:36:30Z" level=info msg="  Waiting for ClusterServiceVersion \"install-osus-here/update-service-operator.v5.0.2-dev\" to appear"
time="2023-08-10T18:36:53Z" level=info msg="  Found ClusterServiceVersion \"install-osus-here/update-service-operator.v5.0.2-dev\" phase: Pending"
time="2023-08-10T18:36:54Z" level=info msg="  Found ClusterServiceVersion \"install-osus-here/update-service-operator.v5.0.2-dev\" phase: Installing"
time="2023-08-10T18:37:01Z" level=info msg="  Found ClusterServiceVersion \"install-osus-here/update-service-operator.v5.0.2-dev\" phase: Succeeded"
time="2023-08-10T18:37:01Z" level=info msg="OLM has successfully installed \"update-service-operator.v5.0.2-dev\""
deployment.apps/updateservice-operator condition met
```

**Review by petr-muller — APPROVED** [review]
/lgtm
/approve
/pj-rehearse ack
/hold

This looks good to me! Contrary to my previous suggestion to only add the install-bundle job temporarily, I am now inclined to keep it to continuously test the ability to install the built bundle, WDYT? In the future we may even extend it to better test that the operator installed this way actually works... If you agree, feel free to just unhold the PR and let it merge ;)

**Comment by petr-muller** [comment]
/retitle OTA-1007 Enable pre-merge test for cincinnati

**Comment by petr-muller** [comment]
/retitle OTA-1007: Enable pre-merge test for cincinnati

**Comment by petr-muller** [comment]
/lgtm cancel

Let me verify one more thing, maybe we'll need to do some renames in our configs (see my [JIRA comment](https://issues.redhat.com/browse/OTA-1007?focusedId=22789615&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22789615))

Specifically, maybe we'll need to _somehow_ rename our operand image in the operator config so that cluster-bot is able to find it:

https://github.com/openshift/release/pull/42053/files#diff-3e4c9ff59bb36d000ea248cbef83680065c76329d216d38ce61c00fb798d3e94R6-R9

**Comment by shellyyang1989** [comment]
> /lgtm cancel
> 
> Let me verify one more thing, maybe we'll need to do some renames in our configs (see my [JIRA comment](https://issues.redhat.com/browse/OTA-1007?focusedId=22789615&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22789615))
> 
> Specifically, maybe we'll need to _somehow_ rename our operand image in the operator config so that cluster-bot is able to find it:
> 
> https://github.com/openshift/release/pull/42053/files#diff-3e4c9ff59bb36d000ea248cbef83680065c76329d216d38ce61c00fb798d3e94R6-R9

I spoke to @jianzhangbjz about it and seems like any image name is fine.

**Comment by petr-muller** [comment]
This looks promising...

https://github.com/openshift/ci-chat-bot/blob/0bd1bfd3ca0294338abfd861863c1e096143b84d/pkg/manager/prow.go#L716

Right, so the match between the names needs to be between the `to` (name of the image) in the operand-building repo and `base_images.[].tag` in the operator one. And there's an implicit match when there's `promotion` and no renaming shenanigans. So this should work. 

My only concern is that our variant config will confuse cluster bot a bit. We promote the `deploy` image built from the `__customrust` variant:

https://github.com/openshift/release/blob/49a3c7ed4ff702ba458bd569ae3dc43e06f3a2cf/ci-operator/config/openshift/cincinnati/openshift-cincinnati-master__customrust.yaml#L44C49-L45

but we also build the same image in the main file:
https://github.com/openshift/release/blob/49a3c7ed4ff702ba458bd569ae3dc43e06f3a2cf/ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml#L40-L45C9

It does not seem cluster-bot is taking variants into account, so maybe the the resulting bundle will inject the "standard" `deploy` file instead of the `__customrust` one. The difference is mainly in Dockerfiles used ([customrust](https://github.com/openshift/cincinnati/blob/master/dist/Dockerfile.deploy/Dockerfile) vs [standard](https://github.com/openshift/cincinnati/blob/master/dist/Dockerfile.rust-toolset/Dockerfile)).

Which is fine, I guess? The point of pre-merge testing is more to test the code, not the productized artifact (which will only be built after merge anyway).

/lgtm

So to me this looks good, feel free to unhold or modify at will (see my previous [comment](https://github.com/openshift/release/pull/42053#pullrequestreview-1573744068) about the hold)

**Comment by shellyyang1989** [comment]
The install-bundle job will not bother us right? I'm fine to leave it as it is. Thank you Petr for your help and support.

/unhold

**Comment by petr-muller** [comment]
```
INFO[2023-08-14T13:26:27Z] Running step install-bundle-openshift-configure-cincinnati. 
INFO[2023-08-14T13:26:35Z] Step install-bundle-openshift-configure-cincinnati succeeded after 8s. 
```

/pj-rehearse ack

**Pull request merged**
